### PR TITLE
Feat: refactor codebase

### PR DESCRIPTION
**Pull request by Hope1226**
In this feature branch we: 

- [x] Separated all methods related to people into a new Class `PersonCreator`
- [x] Separated all methods related to books into a new Class `BookCreator`
- [x] Separated all methods related to rentals into a new Class `ReantalCreator`

### CODE REVIEW

File: book_creator.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  def initialize
+    @book_list = []
+  end
+
+  def create_book
+    print 'Title: '.blue
+    title = gets.chomp
+    print 'Author: '.blue
+    author = gets.chomp
```
Comment (TedLivist) [inline book_creator.rb:14]: - [ ] In keeping with the SOLID rule, let's refactor this `create_book` method. Kindly extract this code requesting user input into a different method.

  **SUGGESTION:** You can have the method return an array with `title` and `author` variables and then destructure the method here in `create_book` using the **[Splat Operator](https://jsarbada.wordpress.com/2019/02/05/destructuring-with-ruby/)**

File: person_creator.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    @people_list = []
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '.blue
+    status = gets.chomp
+    print 'Age: '.blue
+    age = gets.chomp
+    print 'Name: '.blue
+    name = gets.chomp
```
Comment (TedLivist) [inline person_creator.rb:19]: - [ ] In keeping with the Single-responsibility rule, all requests for inputs should be placed in a different method or class whose sole responsibility is to request and validate inputs. Kindly extract these requests for input into another method. After all refactoring, kindly ensure that your app still works the same

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Hope1226 and @Ntwali-Josue

Good job so far!
✔️ Good documentation
✔️ Classes are defined in different files
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
#### _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada:  :champagne:  :confetti_ball: 
Hi @Hope1226  :wave:   :wave: , 

200w 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- Your README file looks very professional :+1: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**